forge/lang/expander.rkt

```diff
@@ -724,6 +748,38 @@
        (syntax/loc stx (begin block.test-decls ...))
        (syntax/loc stx (begin)))]))
 
+
+; PropertyWhereDecl : PROPERTY-TOK Name OF-TOK Name Expr WHERE-TOK TEST-CONSTRUCT*
+(define-syntax (PropertyWhereDecl stx)
+  (syntax-parse stx
+  [((~literal PropertyWhereDecl) 
```

Comment:
Oh .... there's no need to have 2 copies of this pattern.

The syntax-class should declare attributes for everything we need here, the pattern here should be just `pw:PropertyWhereDeclClass`, and then the template here should use `pw.xxx` for everything.

This is a big rewrite, so I can do it after merging.